### DISCUSSION TITLE
Can this package work with Laravel Jetstream?

### DISCUSSION THREAD

**Discussion by bgencacat** [jenssegers/laravel-mongodb#2116] [Q&A] [answered]
Can I use this package with custom user provider and Jetstream?

**Comment by divine** [comment, marked as answer]
Hello,

Have you tried it? If yes, please let us know.

Thanks!

**Reply by bgencacat** [reply]
It works like a charm. But some Jetstream/Fortify features uses UserProvider -> validateCredentials method, others uses hardcoded Hash::check method for verify the password. You may need to override your Hash::check method if you are using different and unsupported encryption type and custom UserProvider together to use all Jetstream/Fortify features.

**Reply by dasjet** [reply]
@bgencacat what about the `Transaction`!!
Getting this error `Call to a member function beginTransaction() on null`

**Reply by bgencacat** [reply]
@dasjet maybe https://github.com/paouriel/laravel-mongodb-transaction-fix

**Reply by Smolevich** [reply]
@dasjet  we can PR with https://github.com/jenssegers/laravel-mongodb/pull/1904 for supporting transactions

**Reply by dasjet** [reply]
Tried, but getting the error from `vendor\laravel\framework\src\Illuminate\Database\Connection.php:458` instead of `vendor/jenssegers/mongodb/src/Jenssegers/Mongodb/Connection.php`.

**Reply by dasjet** [reply]
> It works like a charm.

@Smolevich I thought @bgencacat would have a solution since `Jetstream` worked for him. Thanks

**Reply by divine** [reply]
> > It works like a charm.
> 
> @Smolevich I thought @bgencacat would have a solution since `Jetstream` worked for him. Thanks

Maybe you're doing something wrong? Did you override that function?

**Reply by dasjet** [reply]
@divine I just reinstalled it using `php artisan jetstream:install livewire` but without `--team` option. 
`--team` mandate `Transactions`.

As @bgencacat said now it's work like a charm.
